simplehash/options.go

```diff
@@ -57,8 +57,8 @@ func WithPublicFromPermissioned() HashOption {
 	}
 }
 
-func WithAsConfirmed() HashOption {
+func WithAsCommitted() HashOption {
```

Comment:
Do we still need AsConfirmed?

What happens when a customer wants to update their tier and they get the ability to confirm their events?